### DISCUSSION TITLE
Cannot map PhpStan NullType to ParserNode

### DISCUSSION THREAD

**Discussion by shyim** [rectorphp/rector#7170] [Q&A] [answered]
Hey,

I cannot map a NullType to a ParserNode as it is blocked for Param usages. https://github.com/rectorphp/rector-src/blob/main/packages/PHPStanStaticTypeMapper/TypeMapper/NullTypeMapper.php#L43

Is this intended. I can have

```
<?php

class Foo {
   public function __construct($a = null)
   {
   }
}
```

in my code. Looks like valid for me

**Comment by TomasVotruba** [comment, marked as answer]
Hi, I think the property cannot be null:

```php
public null $name;
```

What exactly is your issue with what rule and code?

**Reply by shyim** [reply]
Ahhh `\Rector\PHPStanStaticTypeMapper\Enum\TypeKind::PARAM()`  refers to public **TYPE** $something; and not to public foo $something = **TYPE**

**Reply by shyim** [reply]
I will use any and it works 😅

**Reply by TomasVotruba** [reply]
Exactly :+1: :laughing:
